### ISSUE TITLE
"Cannot read property 'length' of undefined" while parsing attached GPX file

### ISSUE THREAD

**Issue by thomaschampagne** [sports-alliance/sports-lib#69]
Hey Dimi!

Small issue while parsing file _mytrail_export.gpx_ ([mytrail_export.zip](https://github.com/sports-alliance/sports-lib/files/5335327/mytrail_export.zip)). It produces the following error: `TypeError: Cannot read property 'length' of undefined`

Error is raised at `importer.gpx.ts:49:136`:

```typescript
 const endDate = isActivity ?
          new Date(trackOrRoute.trkseg[trackOrRoute.trkseg.length - 1].trkpt[trackOrRoute.trkseg
[trackOrRoute.trkseg.length - 1].trkpt.length // HERE !
 - 1].time[0]) :
          new Date(startDate.getTime() + samples.length * 1000);
```

Here a jasmine which reproduce the bug on the file.

```typescript
    it('should parse a MyTrail GPX file', done => {

      // Given
      const path = __dirname + '/fixtures/others/mytrail_export.gpx';
      const gpxString = fs.readFileSync(path).toString();

      // When
      const eventInterfacePromise = SportsLib.importFromGPX(gpxString);

      // Then
      eventInterfacePromise.then((event: EventInterface) => {
        expect(event.getFirstActivity().type).toBeDefined();
        done();
      });
```

Not sure how to fix it properly. Let me know if you need more investigation from me.

(File is coming from my elevate app testers)

**Comment by jimmykane** [comment]
Roger that. Let me fix

**Comment by jimmykane** [comment]
The issue is found. 

The gpx contains a trkseg that is empty as the last 
```
    <trkseg>
    </trkseg>
```
I ll see for a solution

**Comment by thomaschampagne** [comment]
Let me know if you need anything from me :)

**Comment by jimmykane** [comment]
Should be fixed in `5.3.0` coming up asap on npm. 

keep me posted if there are any side effects. I ll push this to production from my side for QS asap
